sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -41,6 +44,14 @@ import scala.collection.JavaConverters._
 private[spark] object CosmosConfigNames {
   val AccountEndpoint = "spark.cosmos.accountEndpoint"
   val AccountKey = "spark.cosmos.accountKey"
+  val AuthType = "spark.cosmos.auth.type"
```

Comment:
Maybe we do not need the authType here, we infer from the configs ->
 if masterKey is provided, then use masterKey,
 if servicePrinciple config is provided, then servicePrinciple
 if MSI config is provided, then MSI ....

Reply:
Make it explicit - easier to understand the intend. So, I prefer what you have right now